app/services/past_hearings_fetcher.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class PastHearingsFetcher < ApplicationService
+  attr_reader :prosecution_case_id
+
+  def initialize(prosecution_case_id)
+    @prosecution_case_id = prosecution_case_id
+  end
+
+  def call
+    hearing_summaries = Array(ProsecutionCase.find(prosecution_case_id).body["hearingSummary"]).map do |hearing_summary_data|
+      HmctsCommonPlatform::HearingSummary.new(hearing_summary_data)
+    end
+
+    hearing_summaries.each do |hearing_summary|
+      hearing_summary.hearing_days.each do |hearing_day|
+        response = CommonPlatform::Api::HearingFetcher.call(
+          hearing_id: hearing_summary.id,
+          sitting_day: hearing_day.sitting_day,
+        )
+
+        msg = "[#{Current.request_id}] - "
+
+        if response.success? && response.body.blank?
+          msg += "Past result for hearing ID #{hearing_summary.id} is not available."
+          Rails.logger.info(msg)
+          raise StandardError, msg
+        end
+
+        unless response.success?
+          msg += "Unable to fetch past result of hearing ID #{hearing_summary.id}: Common Platform unresponsive with status code: #{response.status}."
```

Comment:
Sorry, nitpicky but you can ignore this. Since CP can respond with a 403 Forbidden or 503 statusses etc. would unresponsive be ok here? I'm assuming that CP tends to timeout / be unresponsive